src/Voron/Data/DirectAddScope.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using Voron.Data.BTrees;
+
+namespace Voron.Data
+{
+    public unsafe class DirectAddScope : IDisposable
+    {
+        private readonly ITree _tree;
+        private uint _usage;
+        private string _allocationStacktrace = null;
+
```

Comment:
This should be enclosed in ```#if DEBUG``` to avoid allocating that space in the stack when not running in debug or validation.